nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment-api/src/main/java/org/nuxeo/ecm/platform/comment/impl/CommentableDocumentAdapter.java

```diff
@@ -67,11 +72,17 @@ public void removeComment(DocumentModel comment) {
     }
 
     public List<DocumentModel> getComments() {
-        return commentManager.getComments(docModel.getCoreSession(), docModel);
+        return getComments(docModel);
     }
 
     public List<DocumentModel> getComments(DocumentModel parent) {
-        return commentManager.getComments(parent);
+        CoreSession session = docModel.getCoreSession();
+        List<Comment> comments = commentManager.getComments(session, parent.getId());
+        return comments.stream().map(comment -> CoreInstance.doPrivileged(session, s -> {
```

Comment:
Shouldn't we open the privileged session once instead of doing it for each comment?